docs/src/pages/docs/why.mdx

```diff
@@ -0,0 +1,221 @@
+---
+layout: docs
+category: Documentation
+permalink: /docs/why
+next: /docs/why
+---
+import { Tabs } from 'nextra/components'
+
+# Why GraphQL Debugger ?
+
+GraphQL servers are built and interacted with differently to that of other API servers. We found a bunch of pitfalls when using off the shelf tools to observe and debug our GraphQL servers, so we built GraphQL Debugger to solve these problems.
+
+## HTTP Post Body
+
+Most GraphQL implementations use `HTTP` `POST` requests to send queries to the server. The query is sent in the body of the request, for example querying the SpaceX API in the browser: 
+
+```ts
+const body = JSON.stringify({
+  query: "{ missions { description payloads { payload_type }}}",
+});
+
+const response = await fetch("https://api.spacex.land/graphql/", {
+  method: "POST",
+  headers: { "Content-Type": "application/json" },
+  body, // <-- Notice the body
+});
+
+const { data, errors } = await response.json();
+```
+
+What this means for us debuggers and observers is that each request to our server will always be displayed as a `HTTP` `POST` request making it difficult to distinguish or filter between requests.
+
+```sh
+# Logs from a GraphQL server
+HTTP POST /graphql 1 second ago
+HTTP POST /graphql 2 seconds ago
+HTTP POST /graphql 3 seconds ago
+HTTP POST /graphql 4 seconds ago
+HTTP POST /graphql 5 seconds ago
+HTTP POST /graphql 6 seconds ago
+HTTP POST /graphql 7 seconds ago
+HTTP POST /graphql 8 seconds ago
+...
+```
+
+For example, if you were using a REST API, you would expect to see a variety of `HTTP` methods such as `GET`, `POST`, `PUT`, `DELETE` etc, making it easier to debug and distinguish between requests and entity.
+
+```sh
+# Logs from a REST server
+HTTP GET /posts 1 second ago
+HTTP POST /post 2 seconds ago
+HTTP PUT /users 3 seconds ago
+HTTP DELETE /users 4 seconds ago
+HTTP GET /posts 5 seconds ago
+HTTP POST /users 6 seconds ago
+HTTP PUT /post 7 seconds ago
+HTTP DELETE /post 8 seconds ago
+```
+
+## Resolver Fan Out
+
+One GraphQL query can result in many resolver functions being called and subsequently many database queries being executed. This is because GraphQL queries are hierarchical and the data is nested.
+
+
+See the examples of how each query can cause many resolver functions and database queries to be called.
+
+ 
+<Tabs items={['1 database call', '2 database calls', '3 database calls']}>
+  <Tabs.Tab>
+    <pre>
+      ```graphql
+query {
+        user { ## <----- SELECT * FROM users
+          name
+          email
+        }
+}
+```
+    </pre>
+  </Tabs.Tab>
+  <Tabs.Tab>
+    <pre>
```

Comment:
you don't need pre tag

Reply:
Thanks, I thought it was helping with the formatting. 

I am experiencing a slight issue where I have to indent the code blocks inside the tabs very far. 

```
  <Tabs.Tab>
      \`\`\`graphql
query {
        user { ## <----- SELECT * FROM users
          name
          email
          posts { ## <----- SELECT * FROM posts
            title
            body
          }
        }
}
\`\`\`
  </Tabs.Tab>  
```

Is there a way I dont need to do this?

Reply:
<img width="419" alt="Screenshot 2024-01-20 at 10 24 43" src="https://github.com/rocket-connect/graphql-debugger/assets/35999252/96a8a03f-3385-432b-b921-93015bff8c73">
<img width="377" alt="Screenshot 2024-01-20 at 10 25 00" src="https://github.com/rocket-connect/graphql-debugger/assets/35999252/9f2cd0d5-c2fc-4b3a-8ab7-7231a473100b">

Reply:
* https://github.com/rocket-connect/graphql-debugger/pull/206/commits/8457911296c3e867bee0f0628dbdd5f55da0d131